### ISSUE TITLE
[@types/slate] Editor.command misses a function overload

### ISSUE THREAD

**Issue by zaunermax** [DefinitelyTyped/DefinitelyTyped#34867]
@majelbstoat @andykent @JanLoebel @YangusKhan @kalley @Kornil @isubasti @sgreav @jackall3n @benjiro

- [x] I tried using the `@types/slate-react` / `@types/slate` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript

I am mentioning the guys from the `@types/slate` package here, as the `@types/slate-react` package is just referencing the core editor.

The command function (`editorInstance.command()`) has two overloads, one where you pass a string and some arguments to call a command and one other where you can directly pass a handler function (see [here](https://docs.slatejs.org/slate-core/editor#command) also in the example that overload is used)

In the type definitions there is only the one overload where you can pass the string and some arguments.

**Comment by sgreav** [comment]
Please feel free to open a PR correcting this, I haven't personally got time right now but will try and address it in the next few weeks if you are unable to do so in the meantime :)

**Comment by zaunermax** [comment]
I've tried to fix it, but the way the typings work is, that the `slate-react` package will pick the types from the base `import('slate').Editor` which introduces problems as there are properties missing and the two editor types are not compatible. 
When putting a ref onto the React Slate component you get an Editor type from `slate-react` but the types only work with the `import('slate').Editor` type.
What could temporarily fix the problem is to simply add an overload to the base editor like so:
```
command(fn: (...args: any[]) => void, ...args: any[]): Editor
```
which is very ugly but works i guess 🤷‍♂️ 
if that's ok for you, I will open a PR

**Comment by hanstar17** [comment]
Is there any progress to this problem?

**Comment by hanstar17** [comment]
I created a pool request for this issue.

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/36068

**Comment by rbuckton** [comment]
Fixed by #36068.
